### PR TITLE
feat(suite): add anayltics to the device onboarding (seed backup types)

### LINKED ISSUE #12381: Analytics for Multi-share backup
To monitor user interactions with the feature, please add an analytics event that captures:

### Onboarding (Wallet creation and backup step)
<aside>

- [x] Menu Interaction Rate: Track how many users open the select menu versus those who proceed with the default option directly.

- [x] Option Selection Breakdown: Record which options are selected by users who open the menu, including returns to the default setting.

- [x] User Context Data: Capture the type of device and seed card (12/20w) each user is using.

</aside>

### Wallet successfully created screen
image

- [ ] We are already monitoring skipped backups statistically in analytics. Please make sure to keep it functional.

### Settings Multi-share backup creation (”Shamirization”)
Untitled

- [x] To statistically monitor the usage of this feature, please implement an analytics event that captures relevant data. We should be able to check how many of the eligible instances (with compatible devices & backups) will go for this option.

### Create multi-share backup explanative modal window (Nice to have)
<aside>
**Nice to have**

- [x] Modal Closure: Record how many users close the modal, indicating disinterest.

- [x] Learn More Clicks: Track how many users click on the [Learn more] button, which can indicate the need for further information or clarity.

- [x] Create Multi-share Backup Clicks: Monitor how many users click on the [Create multi-share backup] button, which will help assess the effectiveness of the content and its ability to engage users.

</aside>

### Multi-share backup created (success) screen
Untitled (3)

- [x] To statistically monitor the usage of this feature, please implement an analytics event that captures relevant data, such as how many [instances that initiated this flow (Create multi-share backup explanative modal window) ended up successfully completing it.

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/12381

### CODE REVIEW

File: packages/analytics/src/utils.ts [hunk 1]
```diff
@@ -80,6 +80,8 @@ interface ReportEventProps {
 }
 
 export const reportEvent = async ({ type, url, options, retry }: ReportEventProps) => {
+    console.log({ type, url, options, retry });
```
Comment (peter-sanderson) [inline packages/analytics/src/utils.ts]: Dont forget to remove this

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx [hunk 2]
```diff
@@ -18,10 +18,15 @@ import { MultiShareBackupStep5Done } from './MultiShareBackupStep5Done';
...
+import { EventType, analytics } from '@trezor/suite-analytics';
 
 type Steps = 'first-info' | 'second-info' | 'verify-ownership' | 'backup-seed' | 'done';
 
-export const MultiShareBackupModal = ({ onCancel }: { onCancel: () => void }) => {
+export const MultiShareBackupModal = ({
+    onCancel: onCancelFromProps,
+}: {
+    onCancel: () => void;
+}) => {
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx]: `MultiShareBackupModal Props`?

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx [hunk 3]
```diff
@@ -33,6 +38,24 @@ export const MultiShareBackupModal = ({ onCancel }: { onCancel: () => void }) =>
...
+
+    const onCancel = () => {
+        if (step !== 'done') {
+            analytics.report({
+                type: EventType.SettingsMultiShareBackup,
+                payload: { action: 'close-modal' },
+            });
+        }
+
+        onCancelFromProps();
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx:58]: I find this harder to read than keeping  `onCancel` and renaming the wrapper to e.g. `handleCancel`.

**Review by komret — APPROVED** [review]
The URL changes should be a separate commit though 🙏

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx (hunk 3 above)
Comment (peter-sanderson) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx:58]: fixed

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx (hunk 2 above)
Comment (peter-sanderson) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx]: fixed

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx (hunk 3 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx:58]: `onCancelFromProps` should have been just `onCancel` though. Let's change it next time we work on this component.

Comment (peter-sanderson) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/MultiShareBackupModal/MultiShareBackupModal.tsx:58]: sorry fixed here: https://github.com/trezor/trezor-suite/pull/12782

**Pull request merged**